storage/client.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	gax "github.com/googleapis/gax-go/v2"
+)
+
+// storageClient is an internal-only interface designed to separate the
```

Comment:
Add a TODO to note that this is not yet used-- it may be confusing for people trying to read this code otherwise.

Also, should we consider moving the public client constructors in storage.go to this file as well?

Reply:
Done.

> should we consider moving the public client constructors in storage.go to this file as well?

Yeah I think that could be good.

Reply:
I'm going to save moving things for another PR. Will add a TODO.